unittests/CMakeLists.txt

```diff
@@ -44,11 +44,25 @@ set(CUDAQ_RUNTIME_TEST_SOURCES
   common/NoiseModelTester.cpp
   integration/tracer_tester.cpp
   integration/gate_library_tester.cpp
+  dynamics/scalar_ops_simple.cpp
+  dynamics/scalar_ops_arithmetic.cpp
+  dynamics/elementary_ops_simple.cpp
+  dynamics/elementary_ops_arithmetic.cpp
+  dynamics/product_operators_arithmetic.cpp
+  dynamics/test_runge_kutta_time_stepper.cpp
+  dynamics/test_runge_kutta_integrator.cpp
+  dynamics/test_helpers.cpp
+  dynamics/rydberg_hamiltonian.cpp
+  dynamics/test_cudm_helpers.cpp
+  dynamics/test_cudm_state.cpp
+  dynamics/test_cudm_time_stepper.cpp
```

Comment:
These `dynamics` tests are very different from the usual `CUDAQ_RUNTIME_TEST_SOURCES` so we should move them into a separate target. Adding them here will cause these tests to be repeated for all simulator backends.